akka-typed/src/main/scala/akka/typed/internal/TimerSchedulerImpl.scala

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed
+package internal
+
+import scala.concurrent.duration.FiniteDuration
+
+import akka.actor.Cancellable
+import akka.annotation.ApiMayChange
+import akka.annotation.DoNotInherit
+import akka.annotation.InternalApi
+import akka.dispatch.ExecutionContexts
+import akka.typed.ActorRef
+import akka.typed.ActorRef.ActorRefOps
+import akka.typed.javadsl
+import akka.typed.scaladsl
+import akka.typed.scaladsl.ActorContext
+import scala.reflect.ClassTag
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object TimerSchedulerImpl {
+  final case class Timer[T](key: Any, msg: T, repeat: Boolean, generation: Int, task: Cancellable)
+  final case class TimerMsg(key: Any, generation: Int, owner: AnyRef)
+
+  def withTimer[T](factory: TimerSchedulerImpl[T] ⇒ Behavior[T]): Behavior[T] = {
```

Comment:
Ah, sneaky - very good type juggling so we just need to implement it once but is exposed using the right types to each of the DSLs. I also like the `withXxx` style for all kinds of additional features an actor may want 👍